draft-bormann-core-responses.md

```diff
@@ -161,6 +161,73 @@ generated by the client, or in close collaboration with the client (for
 example by the client allowing a third party to use a subset of its
 token values in order to set up non-traditional responses).
 
+# OSCORE processing for non-traditional responses
+
+OSCORE {{oscore}} is built with the assumption that requests are processed into exactly one response.
+The specification contains explicit provisions for Observe requests,
+and a whole protocol extension for multicast requests.
+
+OSCORE's binding between requests and responses remains unmodified:
+Each response is cryptographically bound to an OSCORE request.
+Therefore, any phantom request needs to be an OSCORE request as well,
+and the parties need to agree on the sender and sequence number of the phantom request.
+An easy way to do that securely is to deliver the phantom request in a way the server do the full OSCORE request processing on it.
+The server may process the OSCORE request into internal data structures at reception time,
+or may process it whenever a response is to be sent.
+In the latter case, it may need to relax the requirements of {{Section 8.2 of oscore}} item 3.
+
+To avoid reinventing the same rules as for Observe requests for any other non-traditional response,
+this documents defines a set of processing instructions which can be referenced when specifying their options.
+These rules generalize {{Sections 8.3 and 8.4 of oscore}}:
+
+* In 8.3 step 3, "use the AEAD nonce from the request" is only an option one time
+  after the sequence number expressed in that request was removed from the replay window.
+  This option is usually taken in the first response,
+  necessitating the use encoded Sender Sequence Numbers in later responses.
+  (Non-traditional responses such as Observe that rely on message ordering may require that request's nonce is used either in the first response or not at all).
+  [^maybealwaysfirst]
+
+  <!-- Conveniently, this is obsoleting some text that's rotting away in lwig-oscore. -->
+
+  As a convenient effect, this generalized rule also implies
+  that when a server performs {{Appendix B.1.2 of oscore}},
+  it needs to use its own Partial IV for the nonce
+  (which without this generalized rule necessitated a "MUST" statement in the appendix).
+
+[^maybealwaysfirst]: CA: We could also just mandate the "either the first or never" behavior.
+  I don't know why one would save the efficient response,
```

Comment:
What does "save the efficient response" mean here?

Reply:
Clarified: The one where the OSCORE option is empty.